File: LeetCode/0728_Self_Dividing_Numbers/solution.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function(left, right) {
    let res=[];
    while(left<=right){
        let a=left;
        while(a>0){
            let num=a%10
            if(left%num!=0) break;
            a-=num;
            a/=10;
            if(a==0)res.push (left)
        }
        left++;
    }
    return res
};

File: LeetCode/0728_Self_Dividing_Numbers/solution.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> res;
        while (left <= right) {
            int a = left;
            while (a > 0) {
                int num = a%10;
                if (num == 0 || left % num != 0 ) break;
                a -= num;
                a /= 10;
                if (a == 0)
                    res.push_back(left);
            }
            left++;
        }
        return res;
    }
};